### PR TITLE
[SPARK-45620][PYTHON] Fix user-facing APIs related to Python UDTF to use camelCase

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Fix user-facing APIs related to Python UDTF to use camelCase.

### Why are the changes needed?

To keep the naming convention for user-facing APIs.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Updated the related tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @dtenedor @HyukjinKwon

**Review by dtenedor — APPROVED** [review]
Thanks for the fix!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
Seems there are scala side test failed after this one merged:

- https://github.com/apache/spark/actions/runs/6608112828/job/17946379914
- https://github.com/apache/spark/actions/runs/6608763491/job/17947949938

image

I test `PythonUDTFSuite` locally:

1. before this pr

```
// [SPARK-44753][PYTHON][CONNECT] XML: pyspark sql xml reader writer
git reset --hard 9f675c54a56e8165e24e84a83c186c949ced5be8
build/sbt clean "sql/testOnly org.apache.spark.sql.execution.python.PythonUDTFSuite"
```
then 

```
[info] Run completed in 8 seconds, 301 milliseconds.
[info] Total number of tests run: 9
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 9, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```

2. after this pr

```
// [SPARK-45620][PYTHON] Fix user-facing APIs related to Python UDTF to use camelCase
git reset --hard e3ba9cf0403ade734f87621472088687e533b2cd
build/sbt clean "sql/testOnly org.apache.spark.sql.execution.python.PythonUDTFSuite"
```

then 

```
15:46:02.673 WARN org.apache.spark.sql.catalyst.analysis.SimpleTableFunctionRegistry: The function testudtf replaced a previously registered function.
[info] - SPARK-44503: Specify PARTITION BY and ORDER BY for TABLE arguments *** FAILED *** (420 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: [TABLE_VALUED_FUNCTION_FAILED_TO_ANALYZE_IN_PYTHON] Failed to analyze the Python user defined table function: Traceback (most recent call last):
[info]   File "/Users/yangjie01/SourceCode/git/spark-mine-sbt/python/pyspark/sql/worker/analyze_udtf.py", line 119, in main
[info]     result = handler.analyze(*args, **kwargs)  # type: ignore[attr-defined]
[info]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[info]   File "<string>", line 21, in analyze
[info] AttributeError: 'AnalyzeArgument' object has no attribute 'is_table'
[info]  SQLSTATE: 38000; line 8 pos 5
[info]   at org.apache.spark.sql.errors.QueryCompilationErrors$.tableValuedFunctionFailedToAnalyseInPythonError(QueryCompilationErrors.scala:1985)
[info]   at org.apache.spark.sql.execution.python.UserDefinedPythonTableFunctionAnalyzeRunner.receiveFromPython(UserDefinedPythonFunction.scala:229)
[info]   at org.apache.spark.sql.execution.python.UserDefinedPythonTableFunctionAnalyzeRunner.receiveFromPython(UserDefinedPythonFunction.scala:186)
[info]   at org.apache.spark.sql.execution.python.PythonPlannerRunner.runInPython(PythonPlannerRunner.scala:103)

...
[info] - SPARK-45402: Add UDTF API for 'analyze' to return a buffer to consume on class creation *** FAILED *** (39 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: [TABLE_VALUED_FUNCTION_FAILED_TO_ANALYZE_IN_PYTHON] Failed to analyze the Python user defined table function: Traceback (most recent call last):
[info]   File "/Users/yangjie01/SourceCode/git/spark-mine-sbt/python/pyspark/sql/worker/analyze_udtf.py", line 119, in main
[info]     result = handler.analyze(*args, **kwargs)  # type: ignore[attr-defined]
[info]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[info]   File "<string>", line 16, in analyze
[info] AttributeError: 'AnalyzeArgument' object has no attribute 'data_type'
[info]  SQLSTATE: 38000; line 1 pos 14
...
[info] Run completed in 8 seconds, 26 milliseconds.
[info] Total number of tests run: 9
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 7, failed 2, canceled 0, ignored 0, pending 0
[info] *** 2 TESTS FAILED ***
[error] Failed tests:
[error] 	org.apache.spark.sql.execution.python.PythonUDTFSuite
[error] (sql / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
```

GA of this pr passed seems due to this PR did not touch any Scala code, all tests on the Scala side were skipped.

- https://github.com/apache/spark/actions/runs/6607626526/job/17945278871

image


Could you take a look? Thanks @ueshin

**Comment by HyukjinKwon** [comment]
Mind reverting this? (I'm outside so can't revert now)

**Comment by LuciferYang** [comment]
> Mind reverting this? (I'm outside so can't revert now)

OK, revert this one first

**Comment by HyukjinKwon** [comment]
I pushed some changes

File: sql/core/src/test/scala/org/apache/spark/sql/IntegratedUDFTestUtils.scala [hunk 1]
```diff
@@ -543,7 +543,7 @@ object IntegratedUDFTestUtils extends SQLHelper {
         |    def analyze(initial_count, input_table):
         |        buffer = ""
         |        if initial_count.value is not None:
-        |            assert(not initial_count.is_table)
+        |            assert(not initial_count.isTable)
         |            assert(initial_count.data_type == IntegerType())
```
Comment (LuciferYang) [inline sql/core/src/test/scala/org/apache/spark/sql/IntegratedUDFTestUtils.scala:547]: `data_type` should be `dataType`

Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/IntegratedUDFTestUtils.scala:547]: `with_single_partition`, `partition_by` and `order_by` should also be changed accordingly.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**